riptide-compression/src/main/java/org/zalando/riptide/compression/WrappingHttpOutputMessage.java

```diff
@@ -0,0 +1,44 @@
+package org.zalando.riptide.compression;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpOutputMessage;
+import org.zalando.fauxpas.ThrowingUnaryOperator;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.io.OutputStream;
+
+final class WrappingHttpOutputMessage implements HttpOutputMessage, AutoCloseable {
+
+    private final HttpOutputMessage message;
+    private final ThrowingUnaryOperator<OutputStream, IOException> wrapper;
+    private OutputStream stream;
+
+    WrappingHttpOutputMessage(HttpOutputMessage message, ThrowingUnaryOperator<OutputStream, IOException> wrapper) {
+        this.message = message;
+        this.wrapper = wrapper;
+    }
+
+    @Nonnull
+    @Override
+    public OutputStream getBody() throws IOException {
+        if (stream == null) {
+            stream = wrapper.apply(message.getBody());
```

Comment:
Actually, this whole class is only used within `writeToCompressing`, i.e. it's safe to decorate the stream early on since we can be sure it will be called.

Reply:
Ok, it'll be on your head! :wink: